File: 1338-Reduce Array Size to The Half.py
from collections import Counter
class Solution(object):
    def minSetSize(self, arr):
        """
        :type arr: List[int]
        :rtype: int
        """
        N = len(arr)
        arrCount = Counter(arr)
        
        count = 0
        size = 0
        
        for val in sorted(arrCount.values(), reverse = True):
            count += val
            size += 1
            if count >= N/float(2):
                return size

File: JavaScript/1338-Reduce Array Size to The Half.js
/**
 * @param {number[]} arr
 * @return {number}
 */
var minSetSize = function(arr) {
    const N = arr.length;
    let arrCount = new Map();
    for (let num of arr){
        arrCount.set(num, (arrCount.get(num)||0) + 1 );
    }
    sortedCount = [...arrCount.values()].sort((a, b) => b-a);
    
    let count = 0;
    let ans = 0;
    
    while (count < N/2){
        count += sortedCount[ans];
        ans += 1;
    }
    return ans
};